src/Microsoft.AspNetCore.Mvc.Razor/RazorPageBase.cs

```diff
@@ -275,6 +276,23 @@ public string EndWriteTagHelperAttribute()
             return content;
         }
 
+        public virtual void PushWriter(TextWriter writer)
```

Comment:
Lets start doing things right and make this `protected`. Unless someone else feels differently.